hub-js/src/local/resolver/mutation/update-type-instances.ts

```diff
@@ -0,0 +1,136 @@
+import { cypherMutation } from "neo4j-graphql-js";
```

Comment:
1. Create TI
```graphql
mutation CreateTypeInstances {
  createTypeInstances(
    in: {
     typeInstances: [
      {
        alias: "helm-release"
        typeRef: { path: "cap.type.helm.chart.release", revision: "0.1.0" }
        value: {
          key: "test" # same as it was
        }
        backend: {
          id: "39ea7541-9c9a-46b4-9b87-ff96c49f07b9"
          context: {
            provider: "mock-me"
          }
        }
      }  
    ]
    usesRelations: []
  }
  ) {
    id
    alias
  }
}

```

1. Lock
```graphql
mutation LockTypeInstances($typeInstanceId: ID!) {
  lockTypeInstances(
    in: {
      ids: [$typeInstanceId]
      ownerID: "namespace/name"
    }
  )
}
```

1. Try to update TI
```graphql
mutation UpdateLockedTypeInstance($typeInstanceId: ID!) {
  updateTypeInstances(
    in: [
      {
        id: $typeInstanceId
        ownerID: "namespace/name"
        typeInstance: {
          attributes: [{ path: "cap.attribute.sample", revision: "0.1.0" }]
          value: { name: "Luke Skywalker" }
        }
      }
    ]
  ) {
    id
    resourceVersions {
      ...TypeInstanceResourceVersion
    }
  }
}
```

There is error as expected, but...

1. Get TI

```graphql
query GetTypeInstance($typeInstanceId: ID!) {
  typeInstance(id: $typeInstanceId) {
    ...TypeInstance
  }
}
```

Result:

```graphql
{
  "errors": [
    {
      "message": "/storage_backend.StorageBackend/GetValue NOT_FOUND: TypeInstance \"d4dca069-9b6a-4b25-a5d5-543f3e804c54\" in revision 2 was not found",
      "locations": [
        {
          "line": 283,
          "column": 5
        }
      ],
      "path": [
        "typeInstance",
        "latestResourceVersion",
        "spec",
        "value"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "path": "/storage_backend.StorageBackend/GetValue",
          "code": 5,
          "details": "TypeInstance \"d4dca069-9b6a-4b25-a5d5-543f3e804c54\" in revision 2 was not found",
          "name": "ClientError",
          "stacktrace": [
            "ClientError: /storage_backend.StorageBackend/GetValue NOT_FOUND: TypeInstance \"d4dca069-9b6a-4b25-a5d5-543f3e804c54\" in revision 2 was not found",
            "    at wrapClientError (/Users/pkosiec/repositories/capact/hub-js/node_modules/nice-grpc/lib/client/wrapClientError.js:9:16)",
            "    at Object.callback (/Users/pkosiec/repositories/capact/hub-js/node_modules/nice-grpc/lib/client/createUnaryMethod.js:31:66)",
            "    at Object.onReceiveStatus (/Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/client.js:180:36)",
            "    at Object.onReceiveStatus (/Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:365:141)",
            "    at Object.onReceiveStatus (/Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:328:181)",
            "    at /Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/call-stream.js:182:78",
            "    at processTicksAndRejections (node:internal/process/task_queues:78:11)"
          ]
        }
      }
    },
    {
      "message": "/storage_backend.StorageBackend/GetValue NOT_FOUND: TypeInstance \"d4dca069-9b6a-4b25-a5d5-543f3e804c54\" in revision 2 was not found",
      "locations": [
        {
          "line": 283,
          "column": 5
        }
      ],
      "path": [
        "typeInstance",
        "resourceVersions",
        0,
        "spec",
        "value"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "path": "/storage_backend.StorageBackend/GetValue",
          "code": 5,
          "details": "TypeInstance \"d4dca069-9b6a-4b25-a5d5-543f3e804c54\" in revision 2 was not found",
          "name": "ClientError",
          "stacktrace": [
            "ClientError: /storage_backend.StorageBackend/GetValue NOT_FOUND: TypeInstance \"d4dca069-9b6a-4b25-a5d5-543f3e804c54\" in revision 2 was not found",
            "    at wrapClientError (/Users/pkosiec/repositories/capact/hub-js/node_modules/nice-grpc/lib/client/wrapClientError.js:9:16)",
            "    at Object.callback (/Users/pkosiec/repositories/capact/hub-js/node_modules/nice-grpc/lib/client/createUnaryMethod.js:31:66)",
            "    at Object.onReceiveStatus (/Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/client.js:180:36)",
            "    at Object.onReceiveStatus (/Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:365:141)",
            "    at Object.onReceiveStatus (/Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:328:181)",
            "    at /Users/pkosiec/repositories/capact/hub-js/node_modules/@grpc/grpc-js/build/src/call-stream.js:182:78",
            "    at processTicksAndRejections (node:internal/process/task_queues:78:11)"
          ]
        }
      }
    }
  ],
  "data": {
    "typeInstance": null
  }
}
```

(All fragments are from our examples)

Reply:
> There is error as expected,
not really, you specified a correct ownerID, but the storage backend was not aware of that so the error was returned.

I added a new ownerID property to OnUpdate rpc, to make it aware when data can be safety updated and the err should not be returned.